src/index.ts

```diff
@@ -739,19 +739,25 @@ export class BufferedChangeset implements IChangeset {
     this.trigger(BEFORE_VALIDATION_EVENT, key);
 
     // TODO: Address case when Promise is rejected.
+    // TODO: await RunningValidations before checking the next in line
     if (isPromise(validation)) {
-      this._setIsValidating(key, true);
-
-      return (validation as Promise<ValidationResult>)
-        .then((resolvedValidation: ValidationResult) => {
-          this._setIsValidating(key, false);
-
-          return this._handleValidation(resolvedValidation, { key, value });
-        })
-        .then(result => {
-          this.trigger(AFTER_VALIDATION_EVENT, key);
-          return result;
-        });
+      this._setIsValidating(key, validation as Promise<ValidationResult>);
+
+      let running: RunningValidations = this[RUNNING_VALIDATIONS];
+      let promises = Object.entries(running);
+
+      return Promise.all(promises).then(() => {
```

Comment:
So the idea here is to finish all existing validations before starting another one.  This won't block user interaction since the `key` already has been set on the changeset.  This is just an async path that will update the changes *sometime later.